dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,188 @@
+     * DotStatefulJobTest#testDotStatefulJobIsReallyStateful()} ()}
+     */
+    private class StatefulTestThread implements Runnable {
+
+        private CyclicBarrier cyclicBarrier;
+        private CountDownLatch countDownLatch;
+        private Scheduler scheduler;
+
+        private StatefulTestThread(final CyclicBarrier cyclicBarrier,
+                final CountDownLatch countDownLatch,
+                final Scheduler scheduler) {
+            this.cyclicBarrier = cyclicBarrier;
+            this.countDownLatch = countDownLatch;
+            this.scheduler = scheduler;
+        }
+
+        @Override
+        public void run() {
+            final String thisThreadName = Thread.currentThread().getName();
+
+            try {
+                final JobDetail jobDetail = getJobDetail(thisThreadName);
+
+                final Trigger trigger = new TriggerBuilder().jobDetail(jobDetail)
+                        .triggerGroupName("StatefulTestJob_trigger_group").build();
+
+                //wait for all threads to schedule jobs concurrently
+                cyclicBarrier.await();
+                scheduler.scheduleJob(jobDetail, trigger);
+                countDownLatch.countDown();
+            } catch (Exception e) {
+                throw new DotRuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * Inner class with the logic to be executed by each thread in {@link
+     * DotStatefulJobTest#testTriggerBuilder()}
+     */
+    private class StatefulBuilderTestThread implements Runnable {
+
+        private CyclicBarrier cyclicBarrier;
+        private CountDownLatch countDownLatch;
+        private DotStatefulTestJob dotStatefulTestJob;
+
+        private StatefulBuilderTestThread(final CyclicBarrier cyclicBarrier,
+                final CountDownLatch countDownLatch,
+                final DotStatefulTestJob dotStatefulTestJob) {
+            this.cyclicBarrier = cyclicBarrier;
+            this.countDownLatch = countDownLatch;
+            this.dotStatefulTestJob = dotStatefulTestJob;
+        }
+
+        @Override
+        public void run() {
+            final String thisThreadName = Thread.currentThread().getName();
+
+            try {
+                final JobDetail jd = getJobDetail(thisThreadName);
+                final Trigger trigger = new TriggerBuilder().jobDetail(jd)
+                        .triggerGroupName("StatefulTestJob_trigger_group").build();
+
+                //wait for all threads to schedule jobs concurrently
+                cyclicBarrier.await();
+                TestJobExecutor.execute(dotStatefulTestJob, jd, trigger);
+                countDownLatch.countDown();
+            } catch (Exception e) {
+                throw new DotRuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * Method used to create a job detail instance
+     */
+    private JobDetail getJobDetail(String thisThreadName) {
+        System.out.println("Triggering job for thread " + thisThreadName);
```

Comment:
Codacy found an issue: [System.out.println is used](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5971105)